src/Collapse/utilities/point_cloud_edge_collapse_rips_persistence.cpp

```diff
@@ -78,24 +81,37 @@ int main(int argc, char* argv[]) {
     exit(-1);
   }
 
-  Flag_complex_edge_collapser edge_collapser(
-    boost::adaptors::transform(edges(proximity_graph), [&](auto&&edge){
-      return std::make_tuple(source(edge, proximity_graph),
-                             target(edge, proximity_graph),
-                             get(Gudhi::edge_filtration_t(), proximity_graph, edge));
-      })
-  );
+  auto edges_from_graph = boost::adaptors::transform(edges(proximity_graph), [&](auto&&edge){
+        return std::make_tuple(source(edge, proximity_graph),
+                               target(edge, proximity_graph),
+                               get(Gudhi::edge_filtration_t(), proximity_graph, edge));
+      });
+  std::vector<Filtered_edge> edges_list(edges_from_graph.begin(), edges_from_graph.end());
+  
+  std::vector<Filtered_edge> remaining_edges;
+  for (int iter = 0; iter < edge_collapse_iter_nb; iter++) {
+    Flag_complex_edge_collapser edge_collapser(edges_list);
+    edge_collapser.process_edges(
+      [&remaining_edges](Vertex_handle u, Vertex_handle v, Filtration_value filtration) {
+          // insert the edge
+          remaining_edges.emplace_back(u, v, filtration);
+        });
+    if ((iter + 1) < edge_collapse_iter_nb) {
+      edges_list.clear();
+      std::copy(remaining_edges.begin(), remaining_edges.end(), std::back_inserter(edges_list));
+      remaining_edges.clear();
+    }
```

Comment:
```suggestion
    edges_list = std::move(remaining_edges);
    remaining_edges.clear();
```
No need for copying.
(with the current code, we don't even need remaining_edges, we could reuse edges_list directly, but let's wait until after the interface change, where it may not be possible anymore)

Reply:
Ah, but then you also need to update the following to use edges_list instead of remaining_edges, sorry.

Reply:
Yes, I had it :wink: